lib/pure/times.nim

```diff
@@ -205,32 +224,36 @@ proc fromEpochDay(epochday: int64): tuple[monthday: MonthdayRange, month: Month,
   let mp = (5 * doy + 2) div 153
   let d = doy - (153 * mp + 2) div 5 + 1
   let m = mp + (if mp < 10: 3 else: -9)
-  return (d.MonthdayRange, m.Month, (y + ord(m <= 2)).int)
+  Date(monthday: d.MonthdayRange, month: m.Month, year: (y + ord(m <= 2)).int)
 
-proc getDayOfYear*(monthday: MonthdayRange, month: Month, year: int): YeardayRange {.tags: [], raises: [], benign .} =
-  ## Returns the day of the year.
-  ## Equivalent with ``initDateTime(day, month, year).yearday``.
-  assertValidDate monthday, month, year
+proc yearday*(d: Date): YeardayRange {.tags: [], raises: [], benign .} =
+  ## Returns the day of the year since January 1,
+  ## in the range 0 to 365.
+  assertValidDate d
```

Comment:
`assertValidDate` shouldn't be needed here. When taking proper types like `Date` or `DateTime` as input they can be assumed to be valid.

Reply:
done